### PR TITLE
Clarify usage of `convert_from` in the `bytea` docs

### PR DESCRIPTION
**Pull request by sthm**
### Motivation

Clarify usage of `convert_from` in the `bytea` docs

### CODE REVIEW

**Review by benesch — APPROVED** [review]
@sthm I pushed up a change with a heftier explanation of what makes this all confusing. What do you think?

(Approving since you won't be able to approve your PR, but I want your signoff before I merge!)

**Comment by sthm** [comment]
Excellent, this explains some of the confusions I had around this!

**Pull request merged**
